### ISSUE TITLE
yii\db\command execute() don't handle errors on multiple queries

### ISSUE THREAD

**Issue by TomaszKane** [yiisoft/yii2#14856]
### What steps will reproduce the problem?
Try to import db schema (from mysqldump or PhpMyAdmin schema export) with some error on non-first query. In example, change "int" column type in some table to "intFOO" to cause mysql error. 

Schema:
```mysql
SET FOREIGN_KEY_CHECKS=0;
DROP TABLE IF EXISTS `auth_assignment`;
CREATE TABLE `auth_assignment` (
  `item_name` varchar(64) COLLATE utf8_unicode_ci NOT NULL,
  `user_id` intFOO(10) UNSIGNED NOT NULL,
  `created_at` int(11) DEFAULT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
```

Example code (console action):
```php
public function actionReset()
{
    $schemaFile = 'some/path/schema.sql';
    try {
        $command = Yii::$app->db->createCommand(file_get_contents($schemaFile));
        $command->execute();

        Console::output('Schema imported.');
        return self::EXIT_CODE_NORMAL;
    } catch (\Exception $e) {
        Console::error($e->getMessage());
    }
    return self::EXIT_CODE_ERROR;
}
```

### What is the expected result?

Execution ends with proper exception.

### What do you get instead?

"Schema imported." on stdout - no exception thrown, or error reported.

### Additional info

Read this question and approved answer: [How to check if db execute sql fails?](https://stackoverflow.com/questions/46178560/how-to-check-if-db-execute-sql-fails)

I believe that Yii should handle this case?

| Q                | A
| ---------------- | ---
| Yii version      | 2.0.12
| PHP version      | 7.1.x
| Operating system | Kubuntu 17.04

**Comment by rob006** [comment]
Are you sure that second and third query are executed?

**Comment by cebe** [comment]
dependent on the PDO driver settings it may only support one query. Passing multiple queries to createCommand() works in some cases but that is not documented to work and may not work.

**Comment by rob006** [comment]
This is expected PDO behavior: https://bugs.php.net/bug.php?id=61613

**Comment by rob006** [comment]
Maybe we should introduce something like `executeMultiple()`?

**Comment by cebe** [comment]
> Maybe we should introduce something like executeMultiple()?

that would require parsing of SQL to separate statements. That is not easy to do as we need to fully understand SQL syntax of the corresponding dbms.

**Comment by rob006** [comment]
I mean rather using `PDOStatement::nextRowset()` to iterate through all queries passed to `PDOStatement::execute()` to catch exceptions and get number of rows affected by each of these queries.

See example in linked issue in PHP's bugzilla:

```php
$pdo->beginTransaction();
try {
	$statement = $pdo->prepare($sql);
	$statement->execute();
	while ($statement->nextRowset()) {/* https://bugs.php.net/bug.php?id=61613 */};
	$pdo->commit();
} catch (\PDOException $e) {
	$pdo->rollBack();
	throw $e;
}
```

**Comment by cebe** [comment]
Is the query only executed on calling `nextRowset()`? or is just the error not reported back?

**Comment by TomaszKane** [comment]
@rob006 executed is all queries before the broken one.
BTW, in my case transaction is worthless because MYSQL don't support it on tables create/drop/alter.

@cebe queries are executed, `nextRowset()` just iterate on results and throw PDOException when first result is fail.

**Comment by cebe** [comment]
> @cebe queries are executed, nextRowset() just iterate on results and throw PDOException when first result is fail.

if all queries are executed, we might also want to check further queries for failure, e.g. having the last two queries fail, we'd want an exception reporting all of these.

**Comment by papalapa** [comment]
In continue of our discussion I have one more question. Construction:
```
try {
  Yii::$app->db->pdo->setAttribute(\PDO::ATTR_EMULATE_PREPARES, 1);
  $statement = Yii::$app->db->pdo->prepare($sql);
  $statement->execute();
  while ($statement->nextRowset()) {
    //
  }
} catch (\Exception $e) {
  //
}
```
works fine and throw an exception on first failed sql query, but it is not possible to put MySQL code with 'DELIMITER' - the exception will return:

```SQLSTATE[42000]: Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near 'DELIMITER $$ CREATE DEFINER=`root`@`localhost` PROCEDURE `new_procedure`() BEG' at line 1```.

But I if use Yii::$app->db->createCommand($sql)->execute(); - no exceptions with 'DELIMITER'.
PDO doesn`t understand DELIMITER, and what solution in this case?

**Comment by TomaszKane** [comment]
BTW, does delimiter is really need? See: https://stackoverflow.com/a/9053823/1829368
I have schema dump in my project - after remove delimiters near the triggers all queries was successful and triggers was created.

**Comment by papalapa** [comment]
@TomaszKane, yes, thanks, I'm looking for solution to remove delimiters from queriy now.

**Comment by TomaszKane** [comment]
My schema with delimiters:
```mysql
DELIMITER $$
CREATE TRIGGER `fooCounterCacheAfterDelete` AFTER DELETE ON `foo` FOR EACH ROW BEGIN
    UPDATE user SET foo_count = foo_count - 1 WHERE id = OLD.user_id;
END
$$
DELIMITER ;
```
and import fail.
Schema without delimiter:
```mysql
CREATE TRIGGER `fooCounterCacheAfterDelete` AFTER DELETE ON `foo` FOR EACH ROW BEGIN
  UPDATE user SET foo_count = foo_count - 1 WHERE id = OLD.user_id;
END;
```
works fine.
